packages/labs/eleventy-plugin-lit/src/index.ts

```diff
@@ -14,23 +14,26 @@
 // require() calls for the CommonJS modules we import.
 //
 // See https://github.com/microsoft/TypeScript/issues/43329 and
-// https://github.com/microsoft/TypeScript#22321 for more details.
+// https://github.com/microsoft/TypeScript/issues/22321 for more details.
 
 /* eslint-disable @typescript-eslint/no-var-requires */
 
 const path = require('path') as typeof import('path');
+const {Worker: NodeWorker} =
+  require('worker_threads') as typeof import('worker_threads');
 
 type LitPluginOptions = {
   componentModules?: string[];
+  mode?: 'vm' | 'worker';
 };
 
 module.exports = {
   configFunction: function (
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     eleventyConfig: any,
-    {componentModules}: LitPluginOptions = {}
+    {componentModules, mode = 'vm'}: LitPluginOptions = {}
   ) {
-    if (require('vm').Module === undefined) {
+    if (mode === 'vm' && require('vm').Module === undefined) {
```

Comment:
Might be nice for readability to define two separate functions here, instead of `if` statements within one big function.

This particular check could be moved directly into that function. I think also the `renderModulePath` variable below is only used in the `vm` case too.